### PR TITLE
Disable dropout in evaluation steps (Fixes #1262)

### LINKED ISSUE #1262: Dropout applied when computing test WER
During a run started by a command such as:

```
#!/bin/sh

python -u DeepSpeech.py \
  --train_files ... \
  --dev_files ... \
  --test_files ... \
  --train_batch_size 12 \
  --dev_batch_size 8 \ 
  --test_batch_size 8 \ 
  --epoch 13 \
  --learning_rate 0.0001 \
  --display_step 0 \ 
  --validation_step 1 \ 
  --dropout_rate 0.2367 \
  --default_stddev 0.046875 \
  --checkpoint_step 1 \ 
  --log_level 0 \ 
  --checkpoint_dir "${COMPUTE_KEEP_DIR}" \
  --wer_log_pattern "GLOBAL LOG: logwer('${COMPUTE_ID}', '%s', '%s', %f)"\
  --decoder_library_path "/usr/local/tensorflow1.4/lib/libctc_decoder_with_kenlm.so" \
  "$@"
```

The code will cycle through training and validation jobs before computing a WER based on the test data. However, it reuses the graph which has dropout applied still thus the WER is higher than expected.

This can be verified by comparing the WER at the end of a training run vs that of a test run only with `--dropout 0.0`.

There is a line of code that checks if `optimizer is None` and sets `no_dropout` if so but this will not be triggered during a training run:

https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L640

### PR DESCRIPTION
**Pull request by reuben**
I did this on top of 7c43ebd / #1391 because I'm rebasing the streaming inference branch on top of both these commits, so you'll probably want to review just commit 806ef16 here, or do the reviews for both PRs in a single pass on this PR.

Sorry for the inconvenience, I wish I knew git well enough to make this work from separate branches. Maybe cherry-picking is the answer...

### CODE REVIEW

**Comment by reuben** [comment]
The test failure is a network problem, seems to be unrelated to the code.

File: DeepSpeech.py [hunk 1]
```diff
@@ -666,11 +661,15 @@ def get_tower_results(model_feeder, optimizer):
                     # Retain tower's avg losses
                     tower_avg_losses.append(avg_loss)
 
+    avg_loss = tf.reduce_mean(tower_avg_losses, 0)
```
Comment (tilmankamp) [inline DeepSpeech.py]: Code smell: Name recycling of `avg_loss`.

**Comment by reuben** [comment]
@tilmankamp Thanks, fixed!
